theseus_gui/src/pages/Settings.vue

```diff
@@ -1,7 +1,471 @@
+        <input
+          v-model="settings.custom_env_args"
+          type="text"
+          class="input installation-input"
+          placeholder="ENV1=TEST ENV2=TEST2"
+        />
+      </div>
+      <hr class="card-divider" />
+      <div class="settings-group">
+        <div class="sliders">
+          <span class="slider">
+            Minimum Memory
+            <Slider v-model="settings.memory.minimum" :min="1000" :max="8200" :step="10" />
+          </span>
+          <span class="slider">
+            Maximum Memory
+            <Slider v-model="settings.memory.maximum" :min="1000" :max="8200" :step="10" />
+          </span>
+        </div>
+      </div>
+    </Card>
+    <Card class="settings-card">
+      <h2 class="settings-title">Window Size</h2>
+      <div class="settings-group">
+        <div class="settings-group">
+          <div class="sliders">
+            <span class="slider">
+              Width
+              <Slider v-model="settings.game_resolution[0]" :min="400" :max="2562" :step="2" />
+            </span>
+            <span class="slider">
+              Height
+              <Slider v-model="settings.game_resolution[1]" :min="400" :max="2562" :step="2" />
+            </span>
+          </div>
+        </div>
+      </div>
+    </Card>
+    <Card class="settings-card">
+      <h2 class="settings-title">Launcher Settings</h2>
+      <div class="settings-group">
+        <h3>Console</h3>
+        <div class="toggle-setting">
+          Maximum Concurrent Downloads
+          <input
+            v-model="settings.max_concurrent_downloads"
+            type="number"
+            name="concurrent-downloads"
+            class="concurrent-downloads"
+          />
+        </div>
+      </div>
+    </Card>
+    <Card class="settings-card">
+      <h2 class="settings-title">Commands</h2>
+      <div class="settings-group">
+        <div class="toggle-setting">
+          Pre Launch
+          <input ref="javaArgs" v-model="settings.hooks.pre_launch" type="text" class="input" />
+        </div>
+        <div class="toggle-setting">
+          Wrapper
+          <input ref="javaArgs" v-model="settings.hooks.wrapper" type="text" class="input" />
+        </div>
+        <div class="toggle-setting">
+          Post Launch
+          <input ref="javaArgs" v-model="settings.hooks.post_exit" type="text" class="input" />
+        </div>
+      </div>
+    </Card>
+    <Button color="primary" class="save-btn" @click="saveSettings"><SaveIcon />Save changes</Button>
   </div>
 </template>
+
+<style lang="scss">
+.active {
+  background: var(--color-brand-highlight) !important;
+}
+
+.concurrent-downloads {
+  min-height: 40px;
+  max-width: 4.2rem !important;
+  text-align: center;
+}
+
+.auto-detect-modal {
+  .modal-body {
+    width: 45rem !important;
+    .content {
+      padding: 1rem;
+
+      button {
+        width: 100%;
+        display: flex;
+        align-items: center;
+        justify-content: center;
```

Comment:
is this all needed for the button to display correctly? commenting out all of it makes it display the same

Reply:
I added that style so the buttons would be uniform. See below:

With the CSS:
![image](https://user-images.githubusercontent.com/30800863/233128525-616fcbc7-52b6-4bad-9034-eb86b7732ee7.png)

Without the CSS:
![image](https://user-images.githubusercontent.com/30800863/233128676-6e781349-ac05-4d73-b5a2-bfee7231263d.png)